### DISCUSSION TITLE
geneators for pagy?

### DISCUSSION THREAD

**Discussion by benkoshy** [benkoshy/pagy#10] [Polls]
@ddnexus 

### Pagy Generators

Something like this:

`pagy:rails install --bootstrap`

* Copies pagy initializer.
* requires the bootstrap extra.
* adds backend to application controller.
* adds frontend to a helper.
* provides instructions on where to proceed.

Or: 

`pagy:rails install --javascript`

I haven't thought through the API.

### Why?

* Makes it very very easy for people to install - lowers the hurdle massively.
* as per the rails philosophy.

Against this? What is everyone's thoughts?

**Comment by ddnexus** [comment]
> requires the bootstrap extra.

What's so special with it?

**Reply by benkoshy** [reply]
> What's so special with it?

Not just for bootstrap, any other extra will help:

`--semantic`

to adopt the rails and Steve Krugg philosophy: 

- "Don't make me think"
- sensible defaults.

(Also eliminates a whole set of user support requires because people forgot to require the extra)

**Comment by ddnexus** [comment]
then adding a "for example" would help A LOT! 🤓

**Comment by benkoshy** [comment]
I have noticed this a lot in the help. People forget to `require` the relevant pagy helpers. also could be that they are overwhelmed by the documentation, or they could easily MISS things even though they do read the documentation:

![image](https://user-images.githubusercontent.com/15097447/181159672-7ce662f3-7d64-425d-8290-3e4428a11698.png)

**Comment by ddnexus** [comment]
Sure, that's why we have a "Quick Start" which is not overwhelming, especially in the new doc.
